django/core/cache/backends/filebased.py

```diff
@@ -115,9 +115,11 @@ def clear(self):
     def _is_expired(self, f):
         """
         Take an open cache file `f` and delete it if it's expired.
+        An empty file `f` should be considered expired.
         """
-        exp = pickle.load(f)
-        if exp is not None and exp < time.time():
+        with suppress(EOFError):
+            exp = pickle.load(f)
+        if 'exp' not in locals() or exp is not None and exp < time.time():
```

Comment:
Checking for `'exp'` in `locals()` looks a bit hacky and unconventional to me. Why not simply assign `exp` to `None` in the `EOFError` case?

Either:

```py
exp = None
with suppress(EOFError):
    exp = pickle.load(f)
```

OR

```py
try:
    exp = pickle.load(f)
except:
    exp = None
```

Reply:
That looks a lot better than what I had. I will push a change shortly.